--- php85-pecl-expect-0.4.0/expect.c~	2026-02-19 13:21:55.000000000 +0100
+++ php85-pecl-expect-0.4.0/expect.c	2026-02-19 13:23:40.059226939 +0100
@@ -309,12 +309,13 @@ PHP_FUNCTION(expect_expectl)
 	struct exp_case *ecases, *ecases_ptr, matchedcase;
 #if PHP_MAJOR_VERSION >= 7
     zval *z_stream, *z_cases, *z_match=NULL, *z_case, *z_value;
+	zend_ulong key;
 #else
 	zval *z_stream, *z_cases, *z_match=NULL, **z_case, **z_value;
+	ulong key;
 #endif
 	php_stream *stream;
 	int fd, argc;
-	ulong key;
 	
 	if (ZEND_NUM_ARGS() < 2 || ZEND_NUM_ARGS() > 3) { WRONG_PARAM_COUNT; }
 
--- php73-pecl-expect-0.4.0/expect.c~	2026-02-19 13:50:13.000000000 +0100
+++ php73-pecl-expect-0.4.0/expect.c	2026-02-19 13:53:50.294628963 +0100
@@ -453,11 +453,15 @@ PHP_FUNCTION(expect_expectl)
 		if (z_match && exp_match && exp_match_len > 0) {
 			char *tmp = (char *)emalloc (sizeof(char) * (exp_match_len + 1));
 			strlcpy (tmp, exp_match, exp_match_len + 1);
-#if PHP_MAJOR_VERSION >= 7
+#if PHP_MAJOR_VERSION > 7 || (PHP_MAJOR_VERSION == 7 && PHP_MINOR_VERSION >= 4)
             z_match = zend_try_array_init(z_match);
-			if (!z_match) {
-				return;
-			}
+            if (!z_match) {
+                return;
+            }
+            add_index_string(z_match, 0, tmp);
+#elif PHP_MAJOR_VERSION >= 7
+            zval_dtor(z_match);
+            array_init(z_match);
             add_index_string(z_match, 0, tmp);
 #else
 			zval_dtor (z_match);
--- php56-pecl-expect-0.4.0/expect_fopen_wrapper.c~	2020-01-13 19:27:49.000000000 +0100
+++ php56-pecl-expect-0.4.0/expect_fopen_wrapper.c	2026-02-19 15:25:33.810648390 +0100
@@ -26,6 +26,9 @@
 #if PHP_MAJOR_VERSION >= 7
 php_stream *php_expect_stream_open (php_stream_wrapper *wrapper, const char *command, const char *mode, int options, 
                            zend_string **opened_command, php_stream_context *context STREAMS_DC TSRMLS_DC)
+#elif PHP_MAJOR_VERSION == 5 && PHP_MINOR_VERSION >= 6
+php_stream *php_expect_stream_open (php_stream_wrapper *wrapper, const char *command, const char *mode, int options,
+							  char **opened_command, php_stream_context *context STREAMS_DC TSRMLS_DC)
 #else
 php_stream *php_expect_stream_open (php_stream_wrapper *wrapper, char *command, char *mode, int options, 
 							  char **opened_command, php_stream_context *context STREAMS_DC TSRMLS_DC)
@@ -39,7 +39,7 @@ php_stream *php_expect_stream_open (php_
 		command += sizeof("expect://")-1;
 	} 
 
-#if PHP_MAJOR_VERSION >= 7
+#if PHP_MAJOR_VERSION >= 7 || (PHP_MAJOR_VERSION == 5 && PHP_MINOR_VERSION >= 6)
     if ((fp = exp_popen((char*)command)) != NULL) {
 #else
 	if ((fp = exp_popen(command)) != NULL) {
=======
